tools/actions/live-common-affected/src/main.ts

```diff
@@ -0,0 +1,58 @@
+import { exec } from "child_process";
+import * as core from "@actions/core";
+
+/**
+ * NB: the goal of live-common-affected is to return the main "logical" folders
+ * that are affected by code changes in order to re-run integration tests for them.
+ * This is somehow specific to how live-common works, notably we need to stop at the coin family folder level.
+ */
+
+function main(ref: string) {
+  const cmd = `git diff --name-only ${ref}...HEAD`;
+
+  exec(
+    cmd,
+    {
+      cwd: process.cwd(),
+    },
+    (error, stdout) => {
+      if (error) {
+        core.error(`${error}`);
+        core.setFailed(error);
+        return;
+      }
+      const list = [
+        ...new Set(
+          stdout
+            .split("\n")
+            .map((line) => {
+              const m = line.match(/live-common\/src\/([^/]+)\/([^/]+)/);
+              if (m) {
+                const [, first, second] = m;
+                if (first === "families") {
+                  // in case of coin implementations, we will stop at the coin family level
+                  return `${first}/${second}`;
+                } else {
+                  // in any other case, we consider the first src/* folder level to be relevant filter
+                  return first;
+                }
+              }
```

Comment:
Seems this could be unit tested with different paths.

Reply:
i don't understand 🤔  you mean unit testing the actions? we don't do this today on any of the tools/actions 🤔

Reply:
i have tested the code to work on this PR as well as another PR. all paths were covered that way, we're not having yet unit tests on actions (not even sure it's easy to do, but like we don't unit test our 'github actions' neither anyway, the way to test is by running the workflows)